### PR TITLE
Invoices / Providers section

### LINKED ISSUE #739: Providers section
Propose design. Main elements: 

- stats summary
- summary graph
- providers explorer - I'd go with a client side JS table to explore the table  so we forget about server side paging etc, like
  - https://datatables.net
  - https://mottie.github.io/tablesorter/docs/index.html#Examples
  - https://www.jqwidgets.com/jquery-widgets-demo/demos/jqxgrid/index.htm#demos/jqxgrid/filtering.htm
  - (we should also implement this in admin tables)

Any other element you consider relevant. I'm going to ask for example data.

### LINKED ISSUE #1191: Budgets / Invoices-Providers section
Design in Sketch

- [x] Markup
- [ ] Implementation

### LINKED ISSUE #1314: Invoices DC small glitch when filtering
In Safari at least, the top row moves to bottom when filtering

screencast 2018-01-10 12-55-22

### LINKED ISSUE #1315: Invoices / Improvements to filtering and table
- [x] When filtering the DC, let's update the table
- [x] When filtering the table, let's update the DC
- [x] Table: check if the library has an option out of the box to allow users to customize the number of items per page 
- [x] Table: if you enter a date (ie. "1/1/2017") no results are returned. Maybe the "/" is not correctly scaped?
- [x] Spinner while loading

### PR DESCRIPTION
**Pull request by ferblape**
Closes #1191
Closes #739
Closes #1314 
Closes #1315  

### What does this PR do?

This PR implements a new section for Invoices and Providers in the GobiertoBudgets module.

- This section uses data from PopulateData
- This section needs to be enabled in the configuration of GobiertoBudgets

### How should this be manually tested?

It's enabled in a coule of sites:

- http://getafe.gobify.net/presupuestos/proveedores-facturas

- http://mataro.gobify.net/presupuestos/proveedores-facturas

### Improvements worth to consider

1. ~Add a spinner while the data is being loaded~

2. ~Synchronize DC.js filters with data table, therefore:~
  - ~when a filter is applied in a DC chart the table updates~
  - ~when the table is filtering by a row the charts are updated~

3. Extract `overlay` as a component.

### CODE REVIEW

**Comment by ferblape** [comment]
Markup complete, starting with the integration.

**Comment by ferblape** [comment]
@furilo please review, and give your opinion about the suggested improvements.

**Comment by ferblape** [comment]
@Crashillo look the graphs in Safari and Chrome:

screen shot 2018-01-09 at 18 28 36

**Comment by furilo** [comment]
I'd just review 

- [x] the % of companies/individuals (we need to flip it?)
- [x] "El 0 % de las facturas concentran el 10 % de la facturación" - maybe something is wrong with this calculation?

In order to close this PR. I've created #1315 and #1314 - I'll create other issues for small things.

File: app/assets/stylesheets/comp-graph-dc.scss [hunk 1]
```diff
@@ -64,3 +64,33 @@
     display: inline-block;
   }
 }
+
+//Overlays
+.overlay-container {
```
Comment (Crashillo) [inline app/assets/stylesheets/comp-graph-dc.scss]: This could be turn into a component (partial) to be used all around

Comment (furilo) [reply app/assets/stylesheets/comp-graph-dc.scss]: Yes, go ahead.

**Comment by ferblape** [comment]
Please, merge `master`, because there's a conflict in ` app/assets/stylesheets/application.scss`

**Comment by ferblape** [comment]
I have fixed:

-   the % of companies/individuals (we need to flip it?) > it was wrong, I have flipped it
-  "El 0 % de las facturas concentran el 10 % de la facturación" - maybe something is wrong with this calculation? > the calculation is correct, I have added decimal digits

**Comment by furilo** [comment]
Should we merge this and deploy?

**Comment by Crashillo** [comment]
There are several issues related. Shall we merge now and solve them later (different PRs)?

**Pull request merged**